src/vcpkg/metrics.cpp

```diff
@@ -19,6 +19,69 @@ namespace vcpkg
 {
     LockGuarded<Metrics> g_metrics;
 
+    Optional<StringView> find_first_nonzero_mac(StringView sv)
+    {
+        constexpr static auto is_hex_digit = [](char ch) {
+            return Parse::ParserBase::is_ascii_digit(ch) || (ch >= 'A' && ch <= 'F') || (ch >= 'a' && ch <= 'f');
+        };
+
+        auto first = sv.begin();
+        auto last = sv.end();
+
+        while (first != last)
+        {
+            // XX-XX-XX-XX-XX-XX
+            // 1  2  3  4  5  6
+            // size = 6 * 2 + 5 = 17
+            first = std::find_if(first, last, is_hex_digit);
+            if (last - first < 17)
+            {
+                break;
+            }
+
+            bool is_first = true;
+            bool is_valid = true;
+            auto end_of_mac = first;
+            for (int i = 0; is_valid && i < 6; ++i)
+            {
+                if (!is_first)
+                {
+                    if (*end_of_mac != '-')
+                    {
+                        is_valid = false;
+                        break;
+                    }
+                    ++end_of_mac;
+                }
+                is_first = false;
+
+                if (!is_hex_digit(*end_of_mac))
+                {
+                    is_valid = false;
+                    break;
+                }
+                ++end_of_mac;
+
+                if (!is_hex_digit(*end_of_mac))
+                {
+                    is_valid = false;
+                    break;
+                }
+                ++end_of_mac;
+            }
+            if (is_valid && StringView{first, end_of_mac} != "00-00-00-00-00-00")
```

Comment:
```suggestion
            static constexpr StringLiteral MacZeroes = "00-00-00-00-00-00";
            if (is_valid && StringView{first, end_of_mac} != MacZeroes)
```

To avoid an extra strlen

Reply:
this isn't a strlen; it uses the `char const (&)[N]` constructor for StringLiteral. However, this is still a good choice.